lib/models/user.dart

```diff
@@ -5,7 +5,7 @@ import 'model.dart';
 class User implements Model {
   User({this.id, this.name, this.email, this.photoUrl, this.loginType});
 
-  factory User.fromFirebase(FirebaseUser user, LoginType type) {
+  static User fromFirebase(FirebaseUser user, LoginType type) {
```

Comment:
why?

Reply:
`Use the factory keyword when implementing a constructor that doesn’t always create a new instance of its class. For example, a factory constructor might return an instance from a cache, or it might return an instance of a subtype. Another use case for factory constructors is initializing a final variable using logic that can’t be handled in the initializer list.`

I changed it because it seems like it always returns a new instance of the class?

Reply:
I believe the use cases are not only those specified and it makes more sense for me to use factory constructors instead of static function since it basically constructs the given object, which is supposed to be done by a factory in some other design patterns. Besides, in the style guide, it is also mentioned how it would be better to use constructor over static method based on [this link](https://dart-lang.github.io/linter/lints/prefer_constructors_over_static_methods.html). Perhaps any thoughts on why static methods would be better here?

Reply:
We could just make a normal constructor I guess (not static, not factory).

https://stackoverflow.com/questions/53886304/understanding-factory-constructor-code-example-dart

It seems pretty established in the Dart community that `factory` constructors are for cases in which you don't necessarily create a new instance (where you have cache etc.)

Reply:
I see, okay sounds good to me :)